iohub/convert.py

```diff
@@ -248,6 +248,30 @@ def _gen_chunks(self, input_chunks):
                 f"Chunk type {type(input_chunks)} is not supported."
             )
 
+        # Validate and adjust chunks to be compatible with Dask as they
```

Comment:
I think these are good checks but may be misplaced. Chunks are again reshared below to ensure that they are less than MAX_CHUNK_SIZE. I think these check should happen after? What is the dim size is odd or prime number? How would you adjust chunk size then to meet Dask requirements? Please add tests that this function executes correctly.